### ISSUE TITLE
No ARM64 Enterprise Executor Image Available

### ISSUE THREAD

**Issue by cniessigma** [buildbuddy-io/buildbuddy-helm#86]
Hello! I'm trying to deploy the executor Helm chart into my k8s cluster, with an eye towards configuring several groups of executors based on CPU architecture. Unfortunately, it seems like the default image (at present `gcr.io/flame-public/buildbuddy-executor-enterprise:enterprise-v2.45.0`) does not appear to be compatible with arm64? It backs off restarting the container with

```
exec /tini: exec format error
exec /tini: exec format error
exec /tini: exec format error
```

Is there an arm64 version of the container I can use instead? Or perhaps build my own? I looked at the Dockerfiles in the other repo and I don't feel like there is enough information there to be able to build my own, but perhaps I missed it.

**Comment by sluongng** [comment]
@cniessigma ARM64 is on our current roadmap but there are no pre-built binary/image yet.

Customers who need ARM64 could choose to build the images from the source code directly.
On an ARM64 machine with `docker` installed, you could try the following:

```
git clone https://github.com/buildbuddy-io/buildbuddy
cd buildbuddy

# build the executor binary
bazel build //enterprise/server/cmd/executor

# build the executor image
bazel build //enterprise/server/cmd/executor:executor_image
```

**Comment by cniessigma** [comment]
Fair enough. It's going to be more involved than just that because the Dockerfile for the executor image hard codes every SHA to be the x86_64 of the binaries it downloads, but I will see what we can make do with.

**Comment by siggisim** [comment]
As of yesterday's release (BuildBuddy v2.49.0), our executor image should support running on linux arm64. Please let us know if you run into any issues!

**Comment by cniessigma** [comment]
Amazing thank you!

**Comment by bduffany** [comment]
The containerd issue should be fixed in the latest release (v2.51.0) - let us know if you hit any other issues!
